fedimint-client/src/module/mod.rs

```diff
@@ -17,15 +17,53 @@ use fedimint_core::{
     apply, async_trait_maybe_send, dyn_newtype_define, maybe_add_send_sync, Amount, OutPoint,
     TransactionId,
 };
+use futures::Future;
 
 use crate::sm::{Context, DynContext, DynState, Executor, State};
 use crate::transaction::{ClientInput, ClientOutput};
-use crate::{ClientArc, DynGlobalClientContext};
+use crate::{Client, ClientArc, ClientWeak, DynGlobalClientContext};
 
 pub mod init;
 
 pub type ClientModuleRegistry = ModuleRegistry<DynClientModule>;
 
+#[derive(Clone)]
+pub struct ClientContext {
+    client: Arc<std::sync::RwLock<Option<ClientWeak>>>,
```

Comment:
?

Reply:
Oh, I was looking at the [`tokio` one](https://docs.rs/tokio/latest/tokio/sync/struct.OnceCell.html) which [appears to be `Sync`](https://docs.rs/tokio/latest/tokio/sync/struct.OnceCell.html#impl-Sync-for-OnceCell%3CT%3E).

Reply:
Since we don't do any blocking things in it, it should be OK and more lightweight to use sync ones.

Reply:
After reading more into the implementations of the two I tend to agree. Apparently `tokio::sync::OnceCell` uses a `Semaphore`, which is implemented using a `Mutex` internally. I assumed the more restrictive usage patterns of `OnceCell` would lead to it being less complex internally, but @dpc clearly has the better intuition here.